charts/consul/test/unit/ingress-gateways-deployment.bats

```diff
@@ -1609,4 +1609,30 @@ EOF
   [ "${actual}" = "ca" ]
   local actual=$(echo $object | yq -r '.metadata.annotations."vault.hashicorp.com/ca-cert"')
   [ "${actual}" = "/vault/custom/tls.crt" ]
-}
\ No newline at end of file
+}
+
+#--------------------------------------------------------------------
+# terminationGracePeriodSeconds
+
+@test "ingressGateways/Deployment: terminationGracePeriodSeconds defaults to 10" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -s templates/ingress-gateways-deployment.yaml  \
+      --set 'ingressGateways.enabled=true' \
+      --set 'connectInject.enabled=true' \
+      . | tee /dev/stderr |
+      yq -s -r '.[0].spec.template.spec.terminationGracePeriodSeconds' | tee /dev/stderr)
+  [ "${actual}" = "10" ]
+}
+
+@test "ingressGateways/Deployment: terminationGracePeriodSeconds set defaults to 30" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -s templates/ingress-gateways-deployment.yaml  \
+      --set 'ingressGateways.enabled=true' \
+      --set 'connectInject.enabled=true' \
+      --set 'ingressGateways.defaults.terminationGracePeriodSeconds=30' \
+      . | tee /dev/stderr |
+      yq -s -r '.[0].spec.template.spec.terminationGracePeriodSeconds' | tee /dev/stderr)
+  [ "${actual}" = "30" ]
+}
```

Comment:
I think we need one more test similar to https://github.com/hashicorp/consul-k8s/blob/main/charts/consul/test/unit/ingress-gateways-deployment.bats#L386 to set terminationGracePeriodSeconds on a specific ingress gateway